### PR TITLE
[stable/mysql] #5154 MySql: Allow to override deployment strategy

### LINKED ISSUE #5154: [stable/mysql] Change Deployment strategy to 'recreate'
**FEATURE REQUEST**

**Version of Helm and Kubernetes**:

helm 2.8.2, k8s 1.8.11

**Which chart**:

stable/mysql

**What happened**:

Deployment doesn't specify a strategy, and so defaults to 'spec.strategy: rolling-update' which allows new pods to possibly overlap existing pods. But MySQL should not have two instances sharing the same files. So it seems better/important to specify the 'recreate' strategy.

**What you expected to happen**:

Expect Deployment templates to have 'spec.strategy: recreate'.

### PR DESCRIPTION
**Pull request by janario**
#### What this PR does / why we need it:

* Allow to override mysql deployment strategy
* Change the default to `Recreate`

#### Which issue this PR fixes
  - fixes #5154

#### Special notes for your reviewer:
@olemarkus  @viglesiasce 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ x ] Chart Version bumped
- [ x ] Variables are documented in the README.md
- [ x ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/mysql/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: mysql
-version: 1.4.0
+version: 1.4.1
```
Comment (olemarkus) [inline stable/mysql/Chart.yaml]: You should bump minor version here

**Comment by janario** [comment]
@olemarkus @lachie83 

Any updates here? :)

**Review by olemarkus — CHANGES_REQUESTED** [review]

**Comment by olemarkus** [comment]
Forgot to submit the review. sorry

File: stable/mysql/Chart.yaml (hunk 1 above)
Comment (janario) [reply stable/mysql/Chart.yaml]: I've just pulled master 1.5.0

Do you see it as `1.5.1` or `1.6.0` ?

**Comment by olemarkus** [comment]
New feature means minor version bump. So 1.6

File: stable/mysql/Chart.yaml (hunk 1 above)
Comment (janario) [reply stable/mysql/Chart.yaml]: Done :+1:

**Comment by janario** [comment]
Thanks  🙂✌️

**Pull request merged**